code/modules/mob/living/living.dm

```diff
@@ -1657,13 +1657,13 @@ Thanks.
 				returnToPool(G)
 	if(!item)
 		return FAILED_THROW	//Grab processing has a chance of returning null
+	if(isitem(item))
+		var/obj/item/I = item
+		if(I.cant_drop > 0)
+			to_chat(usr, "<span class='warning'>It's stuck to your hand!</span>")
+			return FAILED_THROW
 
-	var/obj/item/I = item
-	if(istype(I) && I.cant_drop > 0)
-		to_chat(usr, "<span class='warning'>It's stuck to your hand!</span>")
-		return FAILED_THROW
-
-	I.pre_throw()
```

Comment:
It looks like the problem was just the fact that this proc was not within the `if()`, not to do with the typecasting.

Reply:
The if statement would've returned though. This needs to be checked on a potentially succesful throw

Reply:
I agree with Shadowmech. `istype(I)` is a typecheck in & of itself.

Reply:
On what? I've reformatted it. It checks if the item is actually a subtype of item (fucking variable names), then it casts it as an item. It then returns if the item can't drop is > 0, and if not it continues and calls I.pre_throw

Previously it just typecasted the atom as an item regardless of whether it was an atom or not, failed the check as it wasn't an atom, but tried to call pre_throw as if it were an item.

Reply:
I see, the type check and the `cant_drop` check needed to be separated since `pre_throw()` required the former, but not the latter.
That said, however, the typecasting still wasn't really an issue. The same structure with an `istype(I)` instead of instead of an `isitem(item)` would work just as well. Nothing wrong with using `isitem()`, though.

Reply:
But I thought we had holder items for mobs?

Reply:
We do, but the runtime was generated from a player attempting to grab-strangle-throw a `/mob/living/simple_animal/hostile/necro/skeleton`, not picking up a small mob and throwing it from his hand.

Reply:
So why is it not being handled by line 1638?

Reply:
Looked into it a bit more. Deity added `pre_throw` 8 days ago, defined for `/obj/item`. This fix is okay.